### ISSUE TITLE
[bug]: failed funding broadcast doesn't remove pending channels and unlock UTXOs

### ISSUE THREAD

**Issue by Roasbeef** [lightningnetwork/lnd#7868]
### Background

If we fail a broadcast, we don't do anything, and the channel just sits there. This still occupies the locked UTXOs, and also the pending channel entry in the database: https://github.com/lightningnetwork/lnd/blob/03b26bb909bbb8a6b582a5bde3f48c3bf2c4e5c7/funding/manager.go#L2448-L2463

### Steps to reproduce

Double spend a UTXO in a funding transaction. 

### Expected behaviour

UTXO unlocked, and pending channel cleared.

### Actual behaviour

UTXOs still locked, channel shows up in `lncli pendingchannels`.

**Comment by yyforyongyu** [comment]
related to this old PR?
- #5158

**Comment by Roasbeef** [comment]
@yyforyongyu yep, I think let's re-open that PR and also its follow up (#6400)

**Comment by theborakompanioni** [comment]
This recently happened with a peer of mine, after `abandonchannel` **(abandonchannel is a dangerous command and can lead to loss of funds if used incorrectly - ask in our slack channel if unsure)** and a node restart, the channel is gone (as expected), but the UTXOs used for the funding TX are still not credited.

Maybe noteworty: The funding TX contained 4 UTXOs. 2 fully confirmed, one with 0 conf and one that got purged from (probably most) mempools.

Similar to https://github.com/lightningnetwork/lnd/issues/7502 I suppose. What is the suggested procedure here. Would  `lncli sendcoins --sweepall` (as suggested in the linked ticket) pick up the UTXOs and lead to the desired outcome?

**Comment by guggero** [comment]
@theborakompanioni try starting `lnd` with the `reset-wallet-transactions=true` setting (make sure to let it fully sync as a restart won't resume it and leave the wallet in an incomplete state, check the log for progress).

**Comment by theborakompanioni** [comment]
> @theborakompanioni try starting `lnd` with the `reset-wallet-transactions=true` setting (make sure to let it fully sync as a restart won't resume it and leave the wallet in an incomplete state, check the log for progress).

Got a reply from the peer. Unfortunately, that did not do the trick. Change Output from the "unbroadcasted" funding TX is still present in `listunspent` - UTXOs used as inputs for the funding TX not credited :grimacing: 

It feels, according [to the docs](https://github.com/lightningnetwork/lnd/blob/master/docs/recovery.md#forced-in-place-rescan), it should have worked, right? How can it be that the Change Output is still present in `listunspent` after `--reset-wallet-transactions`?

Any other things to try? Feeling a bit bad that I cannot help him.. was really confident that this would work.

**Comment by guggero** [comment]
Are they using Neutrino? Can you paste the output of `listunspent` please?

**Comment by theborakompanioni** [comment]
> Are they using Neutrino? Can you paste the output of `listunspent` please?

I am sorry, I think I am not entitled to publish such private information of another noderunner here.
Wrote you a DM and connected you with the original peer. Thanks @guggero, highly appreciate your help :pray: 
Hopefully some useful insights will come to light that will help you and the project as well.
